examples/contexts/src/main.rs

```diff
@@ -0,0 +1,23 @@
+mod msg_ctx;
+mod producer;
+mod subscriber;
+
+use producer::Producer;
+use subscriber::Subscriber;
+use yew::prelude::*;
+
+use msg_ctx::MessageProvider;
+
+#[function_component]
+pub fn Model() -> Html {
```

Comment:
I would name this `App` and keep  it in a separate `app.rs` file, but up to you m8 :D